smarts/core/sensors.py

```diff
@@ -323,7 +329,46 @@ def _is_done_with_events(cls, sim, agent_id, sensor_state):
         return done, events
 
     @classmethod
-    def _agent_vehicle_is_off_route_and_wrong_way(cls, sim, agent_id):
+    def _agent_reached_goal(cls, sim, vehicle):
```

Comment:
This seems oddly named to me considering it deals with vehicle and mission. Perhaps `_vehicle_reached_mission_goal()`?